shared/actions/unlock-folders.js

```diff
@@ -0,0 +1,39 @@
+/* @flow */
+
+import * as Constants from '../constants/unlock-folders'
+import {routeAppend} from '../actions/router'
+
+import type {RouteAppend} from '../constants/router'
+import type {TypedAsyncAction} from '../constants/types/flux'
+import type {LoadDevices, ToPaperKeyInput, CheckPaperKey, Finish} from '../constants/unlock-folders'
+
+// TODO this won't work in a remote component until we have a synchronous way to get the latest state in a remote component.
+
+function nextPhase (): TypedAsyncAction<RouteAppend> {
+  return (dispatch, getState) => {
+    // TODO careful here since this will not be sync on a remote component!
+    const phase: string = getState().unlockFolders.phase
```

Comment:
So this `nextPhase` is just to have the router reflect the phase of the flow.

It's only purpose is to emit an action to the router (like `routeAppend`) so the route matches the phase.

So it isn't that the UI doesn't match what the phase is until this action is fired, more like the UI _could_ match the router state, and the router state doesn't show the next phase until this action is fired.

**but**, I've thought more about this, and I don't think we should be doing routing (using our Router thing) in remote components. It just adds a bunch of complexity for very little gain.

That means we won't need this `nextPhase` action and the component will reflect it's phase by looking at the phase value in the store.